src/chocolatey/infrastructure/information/ProcessTree.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace chocolatey.infrastructure.information
+{
+    [DebuggerDisplay("{" + nameof(GetDebuggerDisplay) + "(),nq}")]
+    public class ProcessTree
+    {
+        // IGNORED USER AGENT PROCESSES
+        // Our Pester tests may need their own exclusion list in the verification
+        // updated when this list changes. Search the repo for the above string
+        // in caps if you have trouble finding the corresponding list in tests
+        // (should be in UserAgent.Tests.ps1).
+        private static readonly string[] _filteredParents = new[]
+        {
+            // Windows processes and shells
+            "explorer",
+            "winlogon",
+            "powershell",
+            "pwsh",
+            "cmd",
+            "bash",
+            // The name used to launch windows services
+            // in the operating system.
+            "services",
+            "svchost",
+            // Nested processes / invoked by the shim choco.exe
+            "Chocolatey CLI",
+            // Known Terminal Emulators
+            "alacritty",
+            "code",
+            "ConEmu64",
+            "ConEmuC64",
+            "conhost",
+            "c3270",
+            "FireCMD",
+            "Hyper",
+            "SecureCRT",
+            "Tabby",
+            "wezterm",
+            "wezterm-gui",
+            "WindowsTerminal",
+        };
+
+        public ProcessTree(string currentProcessName)
+        {
+            CurrentProcessName = ToFriendlyName(currentProcessName);
+        }
+
+        public string CurrentProcessName { get; }
+
+        public string FirstFilteredProcessName
+        {
+            get { return GetFirstProcess(includeIgnored: false); }
+        }
+
+        public string FirstProcessName
+        {
+            get { return GetFirstProcess(includeIgnored: true); }
+        }
+
+        public string LastFilteredProcessName
+        {
+            get { return GetLastProcess(includeIgnored: false); }
+        }
+
+        public string LastProcessName
+        {
+            get { return GetLastProcess(includeIgnored: true); }
+        }
+
+        public LinkedList<string> Processes { get; } = new LinkedList<string>();
+
+        private static bool IsIgnoredProcess(string value)
+        {
+            return _filteredParents.Contains(value, StringComparer.OrdinalIgnoreCase);
+        }
+
+        protected virtual string ToFriendlyName(string value)
+        {
+            switch (value.ToLowerInvariant())
+            {
+                case "choco":
+                    return "Chocolatey CLI";
+
+                case "ChocolateyGui":

```

Comment:
Should this be lowercase?

Reply:
No, this matches the casing of the `ChocolateyGui.exe` and will be the casing that we expect in the code here unless the user has renamed the executable file.